x-pack/plugins/observability/public/pages/slos_page/hooks/use_fetch_slo_list.ts

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useCallback, useMemo } from 'react';
+import { HttpSetup } from '@kbn/core/public';
+
+import type { SLO, SLOList } from '../../../typings/slo';
+import { useDataFetcher } from '../../../hooks/use_data_fetcher';
+
+const EMPTY_LIST = {
+  results: [],
+  total: 0,
+  page: 0,
+  perPage: 0,
+};
+
+export const useFetchSloList = (): [boolean, SLOList] => {
+  const params = useMemo(() => ({}), []);
+  const shouldExecuteApiCall = useCallback((apiCallParams: {}) => true, []);
+
+  const { loading, data: sloList } = useDataFetcher<{}, SLOList>({
+    paramsForApiCall: params,
+    initialDataState: EMPTY_LIST,
+    executeApiCall: fetchSloList,
+    shouldExecuteApiCall,
+  });
+
+  return [loading, sloList];
+};
+
+const fetchSloList = async (
+  params: {},
+  abortController: AbortController,
+  http: HttpSetup
+): Promise<SLOList> => {
+  try {
+    const response = await http.get<Record<string, unknown>>(`/api/observability/slos`, {
```

Comment:
we can use `SLOS_PAGE_LINK` from `config/paths.ts` in place of `/api/observability/slos`

Reply:
Not sure if we have a configuration for APIs as well, I think I saw different patterns in other plugins like services or something. It would be good to have one place to set APIs paths as well but I don't see it in the scope of this PR.

Reply:
`SLOS_PAGE_LINK` is the app route without the `/api` prefix. But we should maybe have some constant for the API endpoint we are using.

Reply:
Ah, yes, that's right.
I see the paths for some of the alerting APIs are defined in `@kbn/alerting-plugin`. Maybe we do something similar for SLOs in future.